### ISSUE TITLE
Jest: expected/received diff appears twice

### ISSUE THREAD

**Issue by OliverJAsh** [dubzzz/fast-check#3634]
## 🐛 Bug Report

```js
const fc = require("fast-check");

it("works", () => {
  fc.assert(
    fc.property(fc.string(), (a) => {
      expect("foo").toEqual("bar");
    })
  );
});

```

## To Reproduce

In a new directory:
- `yarn add jest fast-check`
- Copy the contents above into `test.js`
- Run `jest`

Result:

```console
$ jest
 FAIL  ./test.js
  ✕ works (7 ms)

  ● works

    Property failed after 1 tests
    { seed: -1070580960, path: "0:0", endOnFailure: true }
    Counterexample: [""]
    Shrunk 1 time(s)
    Got error: Error: expect(received).toEqual(expected) // deep equality

    Expected: "bar"
    Received: "foo"

    Stack trace: Error: expect(received).toEqual(expected) // deep equality

    Expected: "bar"
    Received: "foo"

      4 |   fc.assert(
      5 |     fc.property(fc.string(), (a) => {
    > 6 |       expect("foo").toEqual("bar");
        |                     ^
      7 |     })
      8 |   );
      9 | });

      at toEqual (test.js:6:21)
      at Property.predicate (node_modules/fast-check/lib/check/property/Property.js:17:86)
      at Property.run (node_modules/fast-check/lib/check/property/Property.generic.js:49:33)
      at runIt (node_modules/fast-check/lib/check/runner/Runner.js:21:30)
      at check (node_modules/fast-check/lib/check/runner/Runner.js:73:11)
      at Object.assert (node_modules/fast-check/lib/check/runner/Runner.js:77:17)
      at Object.assert (test.js:4:6)
      Hint: Enable verbose mode in order to have the list of all failing values encountered during the run
      at buildError (node_modules/fast-check/lib/check/runner/utils/RunDetailsFormatter.js:131:15)
      at throwIfFailed (node_modules/fast-check/lib/check/runner/utils/RunDetailsFormatter.js:143:11)
      at reportRunDetails (node_modules/fast-check/lib/check/runner/utils/RunDetailsFormatter.js:156:16)
      at Object.assert (node_modules/fast-check/lib/check/runner/Runner.js:81:52)
      at Object.assert (test.js:4:6)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.198 s, estimated 1 s
Ran all test suites.

```

Observe how the expected/received diff appears twice:

```
    Got error: Error: expect(received).toEqual(expected) // deep equality

    Expected: "bar"
    Received: "foo"

    Stack trace: Error: expect(received).toEqual(expected) // deep equality

    Expected: "bar"
    Received: "foo"
```

## Expected behavior

The expected/received diff should only appear once.

If you have one, please provide a minimal repository reproducing the issue on GitHub.

## Your environment

| Packages / Softwares | Version(s) |
| -------------------- | ---------- |
| fast-check           | 3.6.3 |
| node                 | 16.3.0 |
| TypeScript\*         |            |

\*Only for TypeScript's users

---

This issue seems similar to https://github.com/dubzzz/fast-check/issues/399, although not exactly the same.

**Comment by dubzzz** [comment]
Hey,

It's a partly known issue. If you are using the latest version of Jest, can you give a try to the flag `errorWithCause`.
Now that Jest supports Error with cause, it should be enough to fully fix the problem or at least I hope so.

It can easily be done by running the following code either on top of your file or in the setups file of Jest if you do use one:
```ts
fc.configureGlobal({ errorWithCause: true });
```

**Comment by OliverJAsh** [comment]
This is what I tried:

```diff
 const fc = require("fast-check");
+fc.configureGlobal({ errorWithCause: true });
 
 it("works", () => {
   fc.assert(
     fc.property(fc.string(), (a) => {
       expect("foo").toEqual("bar");
     })
   );
 });

```

Below is the result. Unfortunately it looks like the diff doesn't appear at all now. 🤔 

```console
$ jest
 FAIL  ./test.js
  ✕ works (6 ms)

  ● works

    Property failed after 1 tests
    { seed: -380473213, path: "0:0", endOnFailure: true }
    Counterexample: [""]
    Shrunk 1 time(s)

    Hint: Enable verbose mode in order to have the list of all failing values encountered during the run

      3 |
      4 | it("works", () => {
    > 5 |   fc.assert(
        |      ^
      6 |     fc.property(fc.string(), (a) => {
      7 |       expect("foo").toEqual("bar");
      8 |     })

      at buildError (node_modules/fast-check/lib/check/runner/utils/RunDetailsFormatter.js:134:19)
      at throwIfFailed (node_modules/fast-check/lib/check/runner/utils/RunDetailsFormatter.js:143:11)
      at reportRunDetails (node_modules/fast-check/lib/check/runner/utils/RunDetailsFormatter.js:156:16)
      at Object.assert (node_modules/fast-check/lib/check/runner/Runner.js:81:52)
      at Object.assert (test.js:5:6)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.225 s, estimated 1 s
Ran all test suites.
```

**Comment by dubzzz** [comment]
I'll check on jest side 🤔 In theory starting at 29.4.x they were supposed to properly extract causes linked to errors. Anyway, I'll also recheck the double-print we have on the default option as it's not good.

Seems that the cause is properly forwarded by fast-check as vitest is capable of reading it:
![image](https://user-images.githubusercontent.com/5300235/219850729-e3c51cd6-58c9-4965-b02b-9850c284f816.png)

Same for bare-metal node:
![image](https://user-images.githubusercontent.com/5300235/219850780-85849527-a761-4af6-a7ac-fa8c2f732d61.png)

**Comment by dubzzz** [comment]
For the Error with cause, issue seems to be coming from Jest. I'm currently working on a PR to make it fully supported. For the other issue, I'll attempt a fix on Jest side too. I'll let you know as soon as something gets merged on Jest side.

**Comment by dubzzz** [comment]
A PR attempting to make Error with cause working has been opened on Jest's side, see https://github.com/facebook/jest/pull/13935

**Comment by dubzzz** [comment]
Regarding the remaining issue at the origin the bug report, I mean:

```txt
    Got error: Error: expect(received).toEqual(expected) // deep equality

    Expected: "bar"
    Received: "foo"

    Stack trace: Error: expect(received).toEqual(expected) // deep equality

    Expected: "bar"
    Received: "foo"
```

Actually the bug is fully on fast-check's side 🤔 I'll have to check how I can deal with it properly. It seems that most of the time stack includes the message itself. So printing both implies duplicates.

**Comment by dubzzz** [comment]
Fix for non-cause mode will be included in version 3.7.0 of fast-check
